platform/ios/PolyPodApp/PodApi/Endpoint/Endpoint.swift

```diff
@@ -31,80 +31,67 @@ struct EndpointInfo: Decodable {
 }
 
 final class Endpoint: EndpointProtocol {
-    
+
     init() {
         delegate = nil
     }
-    
+
     let network: Network = Network()
     var delegate: EndpointDelegate?
-    
-    func approveEndpointFetch(endpointId: String, completion: @escaping (Bool) -> Void) {
+
+    private func approveEndpointFetch(endpointId: String, completion: @escaping (Bool) -> Void) {
         delegate?.doHandleApproveEndpointFetch(endpointId: endpointId, completion: completion)
     }
-    
-    private func endpointInfoFromId(endpointId: String) -> EndpointInfo? {
-        let endpointsPath = 
-            Bundle.main.bundleURL.appendingPathComponent("config/endpoints.json")
-        guard let endpointsJsonData = (try? Data(contentsOf: endpointsPath)) else { return nil }
-        guard let endpointsJson = (
-            try? JSONDecoder().decode(
-                Dictionary<String, EndpointInfo>.self, 
-                from: endpointsJsonData
-                )
-            ) else { 
-                return nil 
-            }
-        return endpointsJson[endpointId]
-    }
-    
+
     func send(
-        endpointId: String, 
-        payload: String, 
-        contentType: String?, 
-        authToken: String?, 
+        endpointId: String,
+        payload: String,
+        contentType: String?,
+        authToken: String?,
         completionHandler: @escaping (Error?) -> Void
-        ) {
+    ) {
         approveEndpointFetch(endpointId: endpointId) { approved in
             guard approved else {
-                Log.error("endpoint.send failed: Permission for endpoint \(endpointId) denied")
-                completionHandler(PodApiError.endpointError("send"))
+                Log.error("endpoint.post failed: Permission for endpoint \(endpointId) denied")
```

Comment:
permisison to do what? Post to that endpoint?

Reply:
(it was here already just changed the text) it regards when we check if the endpoint is accessible, when `approveEndpointFetch` is called